tests/semantic_analyser.cpp

```diff
@@ -3741,11 +3741,36 @@ TEST(semantic_analyser, call_offsetof)
           long l; \
         } \
         BEGIN { @x = offsetof(struct Ano, a); }");
+  test("struct Foo { struct Bar { int a; } bar; } \
+        BEGIN { @x = offsetof(struct Foo, bar.a); }");
+  test("struct Foo { struct Bar { int *a; } bar; } \
+        BEGIN { @x = offsetof(struct Foo, bar.a); }");
+  test("struct Foo { struct Bar { struct { int a; } anon; } bar; } \
+        BEGIN { @x = offsetof(struct Foo, bar.anon.a); }");
+  test("struct Foo { struct Bar { struct { int a; }; } bar; } \
+        BEGIN { @x = offsetof(struct Foo, bar.a); }");
+
+  // Error tests
+
+  // Bad type
+  test("struct Foo { struct { int a; } *bar; } \
```

Comment:
Can you use `test_error` and confirm the error messages for these as well? Sorry, I know it's a pain but we're trying to be better about this in our unit tests.